src/AWS.Deploy.Common/UserDeploymentSettings.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.\r
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace AWS.Deploy.Common
+{
+    /// <summary>
+    /// A container for User Deployment Settings file that is supplied with the deployment
+    /// to apply defaults and bypass some user prompts.
+    /// </summary>
+    public class UserDeploymentSettings
+    {
+        public string? AWSProfile { get; set; }
+
+        public string? AWSRegion { get; set; }
+
+        public string? StackName { get; set; }
+
+        public string? SelectedRecipeId { get; set; }
+
+        public JObject? OptionSettingsConfig { get; set; }
+
+        public Dictionary<string, string> OptionSettingItems = new Dictionary<string, string>();
+
+        /// <summary>
+        /// Reads the User Deployment Settings file and deserializes it into a <see cref="UserDeploymentSettings"/> object.
+        /// </summary>
+        /// <exception cref="InvalidUserDeploymentSettingsException">Thrown if an error occured while reading or deserializing the User Deployment Settings file.</exception>
+        public static UserDeploymentSettings? ReadSettings(string filePath)
+        {
+            try
+            {
+                var userDeploymentSettings = JsonConvert.DeserializeObject<UserDeploymentSettings>(File.ReadAllText(filePath));
+                if (userDeploymentSettings.OptionSettingsConfig != null)
+                    userDeploymentSettings.TraverseRootToLeaf(userDeploymentSettings.OptionSettingsConfig.Root);
+                return userDeploymentSettings;
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidUserDeploymentSettingsException("An error occured while trying to deserialize the User Deployment Settings file.", ex);
+            }
+        }
+
+        private void TraverseRootToLeaf(JToken node)
```

Comment:
This logic fails if the user settings has an empty object. For example if there is an empty `ApplicationIAMRole` object the user gets an error the following error:
```
An error occured while trying to deserialize the User Deployment Settings file.
```

Empty objects for complex shapes should be ignored.

Reply:
Also the intent of what is being done here should be commented. It took awhile to reverse engineer this to understand we are initially deseralizing to a Newtonsoft JObject and then post processing after deserialization to create a separate settings collection of just the leaf nodes of the JSON object structure.

Reply:
Added a comment for the method and handled the case when empty objects are encountered.